### PR TITLE
Automated note detection and intaking using computer vision

### PR DESCRIPTION
**Pull request by having11**
Leverages the Limelight 2+ and the Coral TPU accelerator to detect notes, pick the best one, and then automatically path to/intake it

https://github.com/SubZero-Robotics/2024-CommandRobot/assets/16725165/9bfeac83-b4a5-464a-9887-2f9df259db60

### CODE REVIEW

File: src/main/cpp/utils/TargetTracker.cpp [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+      bestTarget.centerX.convert<units::radian>();
+  units::inch_t distance = heightDelta / (tan(verticalAngle.value()));
+  distance -= kDistanceGap;
+
+  frc::Pose2d currentPose = m_drive->GetPose();
+  frc::Rotation2d rotDelta = frc::Rotation2d(bestTarget.centerX);
+
+  auto xTransformation = distance * cos(horizontalAngle.value());
+  auto yTransformation = distance * sin(verticalAngle.value());
+  frc::Transform2d transformDelta =
```
[the remark this answers is not in the document]
Comment (having11) [reply src/main/cpp/utils/TargetTracker.cpp:121]: Should be fixed now

**Review by nobody5050 — COMMENTED** [review]
Will approve when tested on robot and passing ci

File: src/main/include/autos/PathFactory.h [hunk 2]
```diff
@@ -70,7 +75,7 @@ class PathFactory {
     if (ShouldFlip(location)) {
       return pathplanner::AutoBuilder::pathfindToPoseFlipped(
           approxPose,
-          pathplanner::PathConstraints{3.0_mps, 1.5_mps_sq, 540_deg_per_s,
+          pathplanner::PathConstraints{2.5_mps, 3_mps_sq, 540_deg_per_s,
```
Comment (nobody5050) [inline src/main/include/autos/PathFactory.h:78]: Should be constants

File: src/main/include/utils/Commands/DriveCommands.h [hunk 3]
```diff
@@ -25,50 +26,91 @@ namespace DrivingCommands {
...
-        drive->Drive(chassisSpeeds);
-      });
+static RelativeLocation LocationFromProximity(frc::Pose2d currentPose) {
+  auto locationDistances = GetDistancesToFixtures(currentPose);
+
+  auto closestFixture = GetClosestFixture(locationDistances);
+  return closestFixture;
 }
+
+// static frc2::CommandPtr SnapToAngle(DriveSubsystem* drive) {
```
Comment (nobody5050) [inline src/main/include/utils/Commands/DriveCommands.h:96]: Commented out or removed? Should we trim this before merge?

**Review by nobody5050 — CHANGES_REQUESTED** [review]
Very good pr

File: src/main/include/utils/Commands/DriveCommands.h (hunk 3 above)
Comment (having11) [reply src/main/include/utils/Commands/DriveCommands.h:96]: yeah, I can do that

**Pull request merged**